codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAccountIdEndpointModeRuntimeConfig.java

```diff
@@ -0,0 +1,122 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.typescript.codegen;
+
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.rulesengine.traits.EndpointRuleSetTrait;
+import software.amazon.smithy.typescript.codegen.endpointsV2.RuleSetParameterFinder;
+import software.amazon.smithy.typescript.codegen.LanguageTarget;
+import software.amazon.smithy.typescript.codegen.TypeScriptDependency;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.logging.Logger;
+
+/**
+ * Generates accountIdEndpointMode configuration field for service clients that have the eponymous built-in param in the ruleset.
+ */
+@SmithyInternalApi
+public final class AddAccountIdEndpointModeRuntimeConfig implements TypeScriptIntegration {
+
+    private static final Logger LOGGER = Logger.getLogger(AddAccountIdEndpointModeRuntimeConfig.class.getName());
+
+    @Override
+    public void addConfigInterfaceFields(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        if (isAwsService(settings, model)) {
+            ServiceShape service = settings.getService(model);
+            EndpointRuleSetTrait endpointRuleSetTrait = service.getTrait(EndpointRuleSetTrait.class)
+                    .orElseThrow(() -> new RuntimeException("service missing EndpointRuleSetTrait"));
+            RuleSetParameterFinder ruleSetParameterFinder = new RuleSetParameterFinder(service);
+            if (ruleSetParameterFinder.getBuiltInParams().containsKey("AccountIdEndpointMode")) {
+                writer.writeDocs("Defines if the AWS AccountId will be used for endpoint routing.")
+                        .write("accountIdEndpointMode?: AccountIdEndpointMode | __Provider<AccountIdEndpointMode>;\n");
+            }
+        }
+    }
+
+    @Override
+    public void prepareCustomizations(
+        TypeScriptWriter writer,
+        LanguageTarget target,
+        TypeScriptSettings settings,
+        Model model
+    ) {
+        if (isAwsService(settings, model) && target.equals(LanguageTarget.NODE)) {
+            writer.addDependency(AwsDependency.AWS_SDK_CORE);
+            writer.addImport("emitWarningIfUnsupportedVersion", "awsCheckVersion", AwsDependency.AWS_SDK_CORE);
+            writer.write("awsCheckVersion(process.version);");
```

Comment:
just want to confirm importing `/core` here is okay, since a chunk of this work will be placed in `/core` itself

Reply:
outdated question^